tests/expected/function-contract/generic_infinity_recursion.rs

```diff
@@ -0,0 +1,11 @@
+#[kani::requires(x != 0)]
```

Comment:
Please add the copyright header and a quick module documentation explaining what you are testing

Reply:
Implemented as suggested.